### PR TITLE
Add version flag

### LINKED ISSUE #56: Expose version via command line
We bootstrap kubernetes and calico via salt. We do things like this for the `kubectl` app

```
curl -L https://storage.googleapis.com/kubernetes-release/release/v1.1.7/bin/linux/amd64/kubectl -o /usr/bin/kubectl:
  cmd.run:
    - unless: kubectl version | grep Client | grep v1.1.7
    - watch_in:
      - service: kubelet
```

It would be create if the `calico` and `calico-ipam` exposed a way to get the version that is currently installed on the system via like an argument 

```
calico --version
```

or 

```
calico version
```

Thanks!

### PR DESCRIPTION
**Pull request by caseydavenport**
Prints the version of the plugin.

Fixes #56 

Example:
```
gulfstream@gulfstream:~/repos/calico-cni$ ./dist/calico -v
{
  "Commit": "582ff4ed9c2c3552ee82e417f35c8aab7ef00127",
  "Version": "v1.0.2-dev",
  "Branch": "version-flag"
}
```

For now, population of the version tag will be done manually on release.

### CODE REVIEW

File: calico.py [hunk 1]
```diff
@@ -18,26 +18,39 @@
...
 from calico_cni.constants import *
-from ipam import IpamPlugin
 from calico_cni.policy_drivers import ApplyProfileError, get_policy_driver
+from ipam import IpamPlugin
 
 # Logging configuration.
 LOG_FILENAME = "cni.log"
 _log = logging.getLogger("calico_cni")
 
+__doc__ = """
```
Comment (caseydavenport) [inline calico.py]: This is new.

File: calico.py [hunk 2]
```diff
@@ -681,12 +694,20 @@ def main():
 
 
 if __name__ == '__main__': # pragma: no cover
+    # Parse out the provided arguments.
```
Comment (caseydavenport) [inline calico.py]: This is new.

File: ipam.py [hunk 3]
```diff
@@ -17,13 +17,25 @@
...
 from netaddr import IPNetwork
 
 from pycalico.ipam import IPAMClient
-from calico_cni.util import (CniError, parse_cni_args, 
+from calico_cni import __version__
+from calico_cni.util import (CniError, parse_cni_args,
                              configure_logging, print_cni_error)
 from calico_cni.constants import *
 
+__doc__ = """
```
Comment (caseydavenport) [inline ipam.py]: This is new.

File: ipam.py [hunk 4]
```diff
@@ -242,4 +254,12 @@ def main():
 
 
 if __name__ == '__main__': # pragma: no cover
+    # Parse out the provided arguments.
```
Comment (caseydavenport) [inline ipam.py]: This is new.

File: calico_cni/__init__.py [hunk 5]
```diff
@@ -0,0 +1 @@
+__version__ = "v1.0.2-dev"
```
Comment (caseydavenport) [inline calico_cni/__init__.py]: This is new.

**Comment by tomdee** [comment]
LGTM to me. You were going to remove the options_first stuff and maybe update the Makefile to populate the __version__ on build

**Pull request merged**